### PR TITLE
Support language-specific transliteration

### PR DESCRIPTION
**Pull request by BlackbitDevs**
https://github.com/pimcore/pimcore/blob/8d9b9cf0fd1ba65ccf7485eb0be0547493ff267f/lib/Tool/Transliteration.php#L32-L38 pretends to support language-specific transliteration but in fact `$language` does not get used at all.

Before:
`Transliteration::toASCII('schön', 'de')` returns `schon`

After (with this PR):
`Transliteration::toASCII('schön', 'de')` returns `schoen`

Base branch is `10.5` because imho it is a bug when the method pretends to support language-specific transliteration but in fact does not.

### CODE REVIEW

**Comment by brusch** [comment]
Using ICU 67.1-7 it seems that your approach only covers `de`. 
If you look at `transliterator_list_ids()` you will see that there's only `de-ASCII` but no further languages with the `-ASCII` suffix, so I'm not sure if this is really generic enough ...

**Comment by BlackbitDevs** [comment]
Better support `de` than not ;-)
And in case there are new rules added to ICU, it will automatically work in future. 

Am not sure how to support other languages, do you have any insights into ICU rules and how to apply them for ASCII conversion?

**Comment by BlackbitDevs** [comment]
Alternatively we could also use https://github.com/Behat/Transliterator for transliteration. 
How do URL slugs implement this currently?

**Comment by brusch** [comment]
@BlackbitDevs 

> Am not sure how to support other languages, do you have any insights into ICU rules and how to apply them for ASCII conversion?

To be honest, not really, but maybe we could implement a fallback, then transliterates to `Latin` first and then to `ASCII`, I guess that should be possible 🤔

**Comment by brusch** [comment]
Regarding [Behat/Transliterator](https://github.com/Behat/Transliterator) : we had that before (not exactly that package, but I think same origin 😊 ) 
See: https://github.com/pimcore/pimcore/tree/6.9/lib/Tool/Transliteration/Data

It has other drawbacks, I'd prefer to stay on ICU data to be honest.

**Comment by BlackbitDevs** [comment]
> but maybe we could implement a fallback, then transliterates to Latin first and then to ASCII

Isn't this exactly what we currently have?
`transliterator_transliterate('Any-Latin; Latin-ASCII; [^\u001F-\u007f] remove', $value);`
The ICU rule is formatted as \<source language\>-\<target language\>. So in this case it converts any characters to latin characters and afterwards latin to ASCII.

I see that you already discussed about this in https://github.com/pimcore/pimcore/issues/3175#issuecomment-413777442 - to me it is not clear why we don't support `<target language>-ASCII` if available and otherwise fall back to current behaviour.

**Comment by brusch** [comment]
@BlackbitDevs you're right, I'm fine with it, but as it is an improvement I'd target it for `11.x`, can you rebase it? 
Thanks in advance!

File: lib/Tool/Transliteration.php [hunk 1]
```diff
@@ -30,6 +30,10 @@ class Transliteration
      */
     public static function toASCII($value, $language = null)
     {
+        if($language !== null && in_array($language.'-ASCII', transliterator_list_ids())) {
+            return transliterator_transliterate($language.'-ASCII; [^\u001F-\u007f] remove', $value);
+        }
+
         $value = transliterator_transliterate('Any-Latin; Latin-ASCII; [^\u001F-\u007f] remove', $value);
         $value = trim($value);
```
Comment (brusch) [inline lib/Tool/Transliteration.php]:
```suggestion
```

I'd remove that, to be consistent and I don't see any benefit of it ...

**Review by brusch — CHANGES_REQUESTED** [review]

**Comment by BlackbitDevs** [comment]
@brusch I rebased to `11.x`

**Pull request merged**